### PR TITLE
Fixes to executable table layout

### PR DESCRIPTION
**Pull request by vmcj**
Still WIP but some questions on layout

image
Do we think the different tables are improvement or cluther?

Should I re-sort the tables on enabled/disabled. I think given that we have the `search` option there it might make it cleaner.

The code tag looks like an improvement:
image

### CODE REVIEW

File: webapp/src/Controller/Jury/ExecutableController.php [hunk 1]
```diff
@@ -42,18 +44,21 @@ public function __construct(
...
         /** @var Executable[] $executables */
         $executables      = $em->createQueryBuilder()
-            ->select('e as executable, e.execid as execid')
+            ->select('e as executable, e.execid as execid, l.langid')
             ->from(Executable::class, 'e')
+            ->leftJoin(Language::class, 'l', Join::WITH, 'l.compile_executable = e.execid')
             ->addOrderBy('e.type', 'ASC')
             ->addOrderBy('e.execid', 'ASC')
             ->getQuery()->getResult();
+        dump($executables);
```
Comment (meisterT) [inline webapp/src/Controller/Jury/ExecutableController.php:61]: left-over dump

File: webapp/src/Controller/Jury/ExecutableController.php [hunk 2]
```diff
@@ -96,14 +103,29 @@ public function indexAction(Request $request): Response
...
+                'cssclass' => count($e->getLanguages()) || count($e->getProblemsCompare()) || count($e->getProblemsRun()) ? '' : 'disabled',
+            ];
+            $executables_table[] = [
                 'data' => $execdata,
                 'actions' => $execactions,
                 'link' => $this->generateUrl('jury_executable', ['execId' => $e->getExecid()]),
+                'cssclass' => count($e->getLanguages()) || count($e->getProblemsCompare()) || count($e->getProblemsRun()) ? '' : 'disabled',
             ];
         }
+        //dump($executables_tables);
```
Comment (meisterT) [inline webapp/src/Controller/Jury/ExecutableController.php:123]: more dumps

File: webapp/src/Controller/Jury/ExecutableController.php [hunk 3]
```diff
@@ -96,14 +103,29 @@ public function indexAction(Request $request): Response
                 'link' => $this->generateUrl('jury_executable_download', ['execId' => $e->getExecid()])
             ];
 
-            $executables_table[]            = [
+            //dump((bool)count($e->getLanguages()));
```
Comment (meisterT) [inline webapp/src/Controller/Jury/ExecutableController.php:106]: remove dumps

**Comment by meisterT** [comment]
I would try to have two tables, one with used and one unused executables and separating them by icon (and sorting) instead and see what looks better.

**Comment by vmcj** [comment]
image

I've now also labeled executables as disabled when the language they belong to are not allowed to be submitted.

File: webapp/src/Controller/Jury/ExecutableController.php (hunk 2 above)
Comment (vmcj) [reply webapp/src/Controller/Jury/ExecutableController.php]: We have a CI which also flags this, as this was WIP I forgot to remove them (but will when merging).

**Pull request closed without merging**